homeassistant/components/obihai/config_flow.py

```diff
@@ -43,20 +46,59 @@ class ObihaiFlowHandler(ConfigFlow, domain=DOMAIN):
 
     VERSION = 1
 
+    def __init__(self) -> None:
+        """Initialize."""
+        self._host: str | None = None
+
     async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
         """Handle a flow initialized by the user."""
         errors: dict[str, str] = {}
 
         if user_input is not None:
-            self._async_abort_entries_match({CONF_HOST: user_input[CONF_HOST]})
-            if await async_validate_creds(self.hass, user_input):
-                return self.async_create_entry(
-                    title=user_input[CONF_HOST],
-                    data=user_input,
-                )
-            errors["base"] = "cannot_connect"
+            try:
+                ip = gethostbyname(user_input[CONF_HOST])
+                await self.async_set_unique_id(get_mac_address(ip=ip))
```

Comment:
We should avoid the use of `get_mac_address`, it can return the wrong value depending on the network setup of the user. The host could be external to the direct (v)lan of Home Assistant.

Reply:
Instead of using `get_mac_address` here, can you first validate credentials, and after that use the Obihai API to get MAC and abort if there's a unique_id collision?

Reply:
I just uploaded a patch that does this.  Hopefully we are circling on a satisfactory solution...